forkify/src/js/helpers.js

```diff
@@ -0,0 +1,41 @@
+import { TIMEOUT_SEC } from "./config.js";
+
+const timeout = function (s) {
+  return new Promise(function (resolve, reject) {
+    setTimeout(function () {
+      reject(new Error(`Request took too long! Timeout after ${s} second`));
+    }, s * 1000);
+  });
+};
+
+export const getJSON = async function (url) {
+  try {
+    const res = await Promise.race([fetch(url), timeout(TIMEOUT_SEC)]);
+    const data = await res.json();
+
+    if (!res.ok) throw new Error(`${data.message} (${res.status})`);
+    return data;
+  } catch (err) {
+    throw err;
```

Comment:
Catch to throw?
![8552](https://user-images.githubusercontent.com/4405297/120087087-b0261500-c0ed-11eb-86c2-b4eab9197516.gif)